### PR TITLE
feat: Complete fields and methods with `self.` prefixed when inside methods

### LINKED ISSUE #7173: Autocomplete should look at `self` variables/methods when in methods
Given
```rust
struct Example { abcde: f32 }

impl Example {
    fn ex(&self) {
        a // Should bring up autocomplete for "self.abcde"
    }
}

### PR DESCRIPTION
**Pull request by Veykril**
w65NbjkZiG
Closes #7173

### CODE REVIEW

**Comment by Veykril** [comment]
Oh I just realized the call parens complete the receiver type a second time as well now oops

File: crates/ide_completion/src/completions/unqualified_path.rs [hunk 1]
```diff
@@ -45,6 +47,22 @@ pub(crate) fn complete_unqualified_path(acc: &mut Completions, ctx: &CompletionC
             cov_mark::hit!(skip_lifetime_completion);
             return;
         }
+        if let ScopeDef::Local(local) = &res {
```
Comment (flodiebold) [inline crates/ide_completion/src/completions/unqualified_path.rs:50]: this isn't really an "unqualified path" we're completing, so I'd think it'd make more sense to separate it?

**Comment by matklad** [comment]
It feels like we might want to add a feature-flag for this?

**Comment by Veykril** [comment]
Fine by me but what would be the reasoning behind this. To just reduce the amount of completions we offer?

**Comment by matklad** [comment]
Good question!

I think, in general, we should strive to be as configurable as possible.
Abundance of configuration is generally OK, as long as defaults are good.
While most folks roll with default configs, I believe there’s a heavy tail
of people who want to tweak minor detail to perfection. Ability to turn off
annoying IDE features is I think a part of this.

For this particular feature, I can easily imagine someone who finds such
completions confusing, and who would want to disable it just because.

Keep in mind as well that our completion is under-configured: I think it
makes sense to features many existing completions, it’s just that we wrote
them before proper configs were a thing.

**Comment by Veykril** [comment]
Ye okay that makes sense 👍

**Comment by matklad** [comment]
@Veykril when adding a flag, could you also add "configurability" as a cross-cutting concern to the architecture.md?

**Comment by Veykril** [comment]
Will do, though I gotta say I'm having trouble thinking of a name for the flag here

**Comment by matklad** [comment]
Yeah, nothing except selfless comes to mind as the name for this config. Luckily, changing config names is cheap, so we shouldn't worry here.

File: crates/ide_completion/src/completions/unqualified_path.rs [hunk 2]
```diff
@@ -32,6 +33,7 @@ pub(crate) fn complete_unqualified_path(acc: &mut Completions, ctx: &CompletionC
         });
         return;
     }
+
```
Comment (matklad) [inline crates/ide_completion/src/completions/unqualified_path.rs:36]: :+1: always nice to give the code some air to breathe!

**Review by matklad — APPROVED** [review]
LGTM!

**Comment by Veykril** [comment]
bors r=matklad

**Pull request merged**